src/Middleware/SpaServices.Extensions/src/DevelopmentServer/DevelopmentServerMiddlewareExtensions.cs

```diff
@@ -0,0 +1,87 @@
+        /// <param name="serverName">The name of the Server used in the Console.</param>
+        public static void UseDevelopmentServer(
+            this ISpaBuilder spaBuilder,
+            string npmScript,
+            string waitText,
+            string serverName = "App")
+        {
+
+            if (string.IsNullOrEmpty(waitText))
+            {
+                throw new InvalidOperationException($"To use {nameof(UseDevelopmentServer)}, you must supply a non-empty value for the {nameof(waitText)} parameter. This allows us the find when the Development Server has started.");
+            }
+
+            if (spaBuilder == null)
+            {
+                throw new ArgumentNullException(nameof(spaBuilder));
+            }
+
+            var spaOptions = spaBuilder.Options;
+
+            if (string.IsNullOrEmpty(spaOptions.SourcePath))
+            {
+                throw new InvalidOperationException($"To use {nameof(UseDevelopmentServer)}, you must supply a non-empty value for the {nameof(SpaOptions.SourcePath)} property of {nameof(SpaOptions)} when calling {nameof(SpaApplicationBuilderExtensions.UseSpa)}.");
+            }
+
+            DevelopmentServerMiddleware.Attach(spaBuilder, npmScript, waitText, serverName);
+        }
+
+        /// <summary>
+        /// Handles requests by passing them through to an instance of the create-react-app server.
+        /// This means you can always serve up-to-date CLI-built resources without having
+        /// to run the create-react-app server manually.
+        ///
+        /// This feature should only be used in development. For production deployments, be
+        /// sure not to enable the create-react-app server.
+        /// </summary>
+        /// <param name="spaBuilder">The <see cref="ISpaBuilder"/>.</param>
+        /// <param name="npmScript">The name of the script in your package.json file that launches the create-react-app server.</param>
+        public static void UseReactDevelopmentServer(
+            this ISpaBuilder spaBuilder,
+            string npmScript)
+        {
+            UseDevelopmentServer(spaBuilder, npmScript, "Starting the development server", "create-react-app");
+        }
+
+        /// <summary>
+        /// Handles requests by passing them through to an instance of the vue-cli-service server.
+        /// This means you can always serve up-to-date CLI-built resources without having
+        /// to run the vue-cli-service server manually.
+        ///
+        /// This feature should only be used in development. For production deployments, be
+        /// sure not to enable the vue-cli-service server.
+        /// </summary>
+        /// <param name="spaBuilder">The <see cref="ISpaBuilder"/>.</param>
+        /// <param name="npmScript">The name of the script in your package.json file that launches the vue-cli-service server.</param>
+        public static void UseVueDevelopmentServer(
```

Comment:
We haven't announced any plan to build in support for Vue-specific things. Previously, we explicitly stepped away from doing that, so we could focus on the two most demanded third-party SPA frameworks (Angular and React). Changing this plan would be a decision for @danroth27.